DOCS-3843 Using RUM & Session Replay for Product Analytics Guide (#14889)
DataDog/documentation: content/en/real_user_monitoring/guide/rum-for-product-analytics.md

---

> title: Use RUM & Session Replay for Product Analytics

Reviewer: can this also be the title that's shown in the nav? felt like the one in the preview was a bit too long

- link: '/real_user_monitoring/explorer/'
  tag: 'Documentation'
  text: 'Learn about the RUM Explorer'
- link: '/real_user_monitoring/'
  tag: 'Documentation'
  text: 'Learn how to visualize your RUM data'
- link: '/real_user_monitoring/frustration_signals'
  tag: 'Documentation'
  text: 'Learn about frustration signals'
- link: '/real_user_monitoring/session_replay'
  tag: 'Documentation'
  text: 'Learn about Session Replay'
- link: '/dashboards/guide/powerpacks-best-practices/'
  tag: 'Documentation'
  text: 'Learn about powerpacks'
---

## Overview

[RUM & Session Replay][1] allows you to monitor trends in consumer behavior and uncover answers to product usage questions about your web and mobile applications.

This guide discusses several use cases to enrich your RUM & Session Replay data and provides example queries you can use to answer questions about product analytics.

## Setup

Once you have set up the Datadog RUM SDK, enrich your [browser][2] or mobile ([iOS][3] and [Android][4]) data with attributes to customize the data according to your use case. For example, adding contextual information allows you to [identify sessions that are tied to specific users][4].

> ## Monitor page traffic and button usage

Reviewer: instead of button let's go with feature

Author: ```suggestion
## Monitor page traffic and feature usage
```

If you are interested in learning what buttons your users click on the most, you can track page traffic and the usage of buttons in your application. 

1. Navigate to the [RUM Explorer][5] and select **Actions** from the dropdown menu next to the search query. 
2. Enter `@view.name:/cart` in the search query and select the **Top List** visualization type.
3. In the `by` field of the `Group into fields` section above, select **Action Name** from the group dropdown.

This example displays the top actions on Shopist's `/cart` page.

{{< img src="real_user_monitoring/guide/rum-for-product-analytics/actions_in_cart_page.png" alt="Search query for actions on Shopist's Cart page" style="width:90%;">}}

> For example: 

Reviewer: in the preview link, i think this is in the wrong spot. you want it before the screenshot no? i suggest moving the "for example" to line 42

> Import this into a dashboard by copying the code snippet and selecting **Import dashboard JSON**: 

Reviewer 2: For this and the other instructions like it... 

I clicked Copy on the code blocks, went to a new dashboard, and from the Settings menu, selected **Import dashboard JSON**, but got an error. 

What did work for me was, after creating the toplist in the Explorer, clicking Export > Export to dashboard and selecting a db to put it in. 

But note that this only works on the Demo org -- which I don't think customers have access to. Even setting up the query you describe in step 2 doesn't work outside Demo -- you'd have to tell them to choose a view name in their particular web app.

So I think the instructions in this guide have to be generalized a bit to clarify when you're telling them to do something exactly, vs when they'll see something slightly different in their actual org.

> {{< code-block lang="json" filename="json" disable_copy="false" collapsible="true" >}}

Reviewer: you were going to add something here to introduce the JSON snippet right? can we please add a line before this one to say something like "if you want to import the JSON snippet to a dashboard, see the following"

                    "group_by": [
                        {
                            "facet": "@action.name",
                            "limit": 5,
                            "sort": {
                                "aggregation": "count",
                                "order": "desc"
                            }
                        },
                        {
                            "facet": "@usr.name",
                            "limit": 5,
                            "sort": {
                                "aggregation": "count",
                                "order": "desc"
                            }
                        }
                    ]
                }
            ],
            "formulas": [
                {
                    "formula": "query1",
                    "limit": {
                        "count": 25,
                        "order": "desc"
                    }
                }
            ]
        }
    ],
    "has_search_bar": "auto"
}
{{< /code-block >}}

## Analyze the conversion rate of core workflows

Use the [funnel visualization type][6] to track the conversion rate across crucial areas in your website. 

> Once you have created a funnel based on views and actions on your website, you can do the following actions:

Reviewer: let's say "views or actions" since someone could build a funnel entirely off views or entirely off actions

Author: ```suggestion
Once you have created a funnel based on views or actions on your website, you can do the following actions:
```

> Once you have created a funnel based on views and actions on your website, you can do the following actions:

Reviewer: and, the last action on line 156 is confusing bc we are talking about actions in terms of the funnel earlier and then things we are saying. can we use a different ending word?

Author: ```suggestion
Once you have created a funnel based on views and actions on your website, you can use it in the following ways:
```

> * Create a [saved view][7] to reference it in the RUM Explorer 

Reviewer: "to reference it in RUM"
 note that people don't only have to reference it from the explorer

Author: ```suggestion
* Create a [saved view][7] to reference it in the RUM & Session Replay
```

Reviewer: wouldn't be "the" RUM but just RUM. no need fo session replay here. "Create a saved view to reference it throughout RUM"

                "query_string": "@type:session",
                "data_source": "rum",
                "steps": [
                    {
                        "facet": "@view.name",
                        "value": "/"
                    },
                    {
                        "facet": "@view.name",
                        "value": "/department/chairs"
                    },
                    {
                        "facet": "@view.name",
                        "value": "/cart"
                    }
                ]
            },
            "request_type": "funnel"
        }
    ]
}
{{< /code-block >}}

### View funnel analysis

The side panel contains detailed information about the load time of an individual view, the conversion and drop off rates based on the country, device type, browser, and version, and outstanding [issues][9] that occurred on the page.

{{< img src="real_user_monitoring/guide/rum-for-product-analytics/funnel_analysis_side_panel.mp4" alt="Funnel Analysis side panel displaying conversion rates, drop off rates, page performance, errors, and user behavior" width=90%; video="true" >}}

## Identify your most frustrated users

> [Frustration Signals][10] surface moments where users express frustrated behavior (a rage click, dead click, or error click) and you can address the most pressing issues that users are facing. 
> 
> When examining user behavior, you want to comprehend the areas of your website where users are getting stuck. 

Reviewer 2: ```suggestion
[Frustration Signals][10] surface moments where users express frustrated behavior (a rage click, dead click, or error click) so you can address the most pressing issues that users are facing. Examining user behavior to identify the areas of your website where users are getting stuck. 
```

                {
                    "formula": "query1",
                    "limit": {
                        "count": 100,
                        "order": "desc"
                    }
                }
            ]
        }
    ],
    "has_search_bar": "auto"
}
{{< /code-block >}}

## Watch the user experience in Session Replay

You can visually watch the impact that poor user experiences have on your users. For example, if you have built a funnel and noticed that the drop off rate is exceptionally high between steps, you can watch a [sesion replay recording][11] to see what a user did before they dropped off. 

In a funnel visualization, you can access the **Funnel Analysis** side panel and click **Sample Replay Session** on sessions where users continued onto another step or dropped off.

{{< img src="real_user_monitoring/guide/rum-for-product-analytics/funnel_sample_session_replay.mp4" alt="The Funnel Analysis side panel contains links to Session Replay" width=90%; video="true" >}}

With Session Replay, you can identify what parts of your product are confusing to users and need improving in order to drive up conversion. 

## Track usage patterns in powerpacks

> [Powerpacks][12] are templated groups of widgets for common monitoring patterns and product analytics. 

Reviewer 2: ```suggestion
[Powerpacks][12] are templated groups of dashboard widgets for common monitoring patterns and product analytics. 
```
